From b19a3d18c62f4ea1ed4b720014e984bf301701f6 Mon Sep 17 00:00:00 2001
From: Liu Ying <b17645@freescale.com>
Date: Wed, 7 Apr 2010 11:41:36 -0400
Subject: [PATCH 105/280] ENGR00122226 IPU lib:Fix potential dead lock issue
 in Screenlayer lib

The function UpdateScreenLayer() in Screenlayer library is likely to
miss to release a semaphore when error occurs,
which may cause dead lock issue.

Signed-off-by: Liu Ying <b17645@freescale.com>
---
 screenlayer/ScreenLayer.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/screenlayer/ScreenLayer.c b/screenlayer/ScreenLayer.c
index 2708428..b97a02a 100644
--- a/screenlayer/ScreenLayer.c
+++ b/screenlayer/ScreenLayer.c
@@ -1252,10 +1252,9 @@ SLRetCode UpdateScreenLayer(ScreenLayer *pSL)
 	}
 
 	ret = _UpdateFramebuffer(pCurSLPriv);
+done:
 	sem_post(semDispID);
 	sem_post(semMainID);
-
-done:
 	return ret;
 }
 
-- 
1.8.0

